### PR TITLE
Add example and documentation for Flake Id Generator

### PR DESCRIPTION
**Pull request by ihsandemir**
Flake Id Generator example and documentation is added.

### CODE REVIEW

File: examples/distributed-primitives/idgenerator/FlakeIdGenerator.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#include <hazelcast/client/HazelcastClient.h>
+#include <hazelcast/client/FlakeIdGenerator.h>
+
+int main() {
+    hazelcast::client::ClientConfig config;
+    hazelcast::client::HazelcastClient hz(config);
+
+    hazelcast::client::FlakeIdGenerator generator = hz.getFlakeIdGenerator("flakeIdGenerator");
+    for (int i = 0; i < 10000; ++i) {
+        hazelcast::util::sleep(1);
```
Comment (SaitTalhaNisanci) [inline examples/distributed-primitives/idgenerator/FlakeIdGenerator.cpp]: Is this in milliseconds ?

Comment (ihsandemir) [reply examples/distributed-primitives/idgenerator/FlakeIdGenerator.cpp]: that is seconds. sleepmillis for milliseconds.

**Comment by SaitTalhaNisanci** [comment]
Why does this PR has `Estimation: L` label? Do we consider this PR with its implementation and is that why it has this label?

**Comment by ihsandemir** [comment]
@SaitTalhaNisanci Corrected estimation to S.

**Pull request merged**